src/string_utils.cpp

```diff
@@ -120,3 +120,22 @@ impl::LineSplitter::Iter impl::LineSplitter::Iter::operator++() noexcept {
     *this = Iter(rest_, false);
     return prev;
 }
+
+std::string lower_case(std::string_view str) {
+    std::string result;
+    result.reserve(str.size());
+    for (auto c : str) {
+        result.push_back(tolower(c));
+    }
+    return result;
+}
+
+bool has_prefix(std::string_view haystack, std::string_view needle) {
+    if (needle.size() > haystack.size())
```

Comment:
I _think_ this is the same as:

```
return haystack.find(needle) == 0;
```

right? or:

```
return needle.size() <= haystack.size() || haystack.substr(0, needle.size() != needle;
```

Either way, the loop here I think can be replaced with a string funciont. C++ doesn't have many (as you've noticed)... but there _are_ some...

Reply:
Yeah, actually I'd rather do the latter. 'find' is way more expensive*. Done.

* yeah. Noticably so. Sure.

Reply:
Neat! find does look for it _anywhere_ and I suspect the compiler isn't smart enough to work out it can stop early. (if only there was some interactive website where we could find out...)